addons/mail/static/src/core/common/message.js

```diff
@@ -395,7 +387,11 @@ export class Message extends Component {
     }
 
     exitEditMode() {
-        this.props.messageEdition.exitEditMode();
+        const threadAsInEdition = this.props.message.threadAsInEdition;
+        this.props.message.composer = undefined;
+        if (threadAsInEdition && threadAsInEdition.eq(this.props.thread)) {
+            threadAsInEdition.composer.autofocus++;
+        }
```

Comment:
This should also be inside the message model.

Reply:
I moved the `exitEditMode` to models, but I still have to pass `thread` as param to let it know the contextual thread being viewed while starting/stopping the edition. I put a docstring on the functions in models to clarify what the `thread` param means.